src/handlers/interactionErrorHandler.ts

```diff
@@ -21,82 +22,74 @@ export const handleError = async (
-        switch (interaction.replied) {
-            case true:
-                logger.warn(
-                    error,
-                    `Interaction '${interactionIdentifier}' threw an error but has already been replied to.`
-                );
-                return;
-            case false:
-                logger.debug('Responding with error embed');
-                return await interaction.editReply(errorReply);
-        }
-    } else if (interaction instanceof MessageComponentInteraction) {
-        switch (interaction.replied) {
-            case true:
-                logger.warn(
-                    error,
-                    `Interaction '${interactionIdentifier}' threw an error but has already been replied to.`
-                );
+        if (
+            (interaction instanceof ChatInputCommandInteraction ||
+                interaction instanceof MessageComponentInteraction) &&
+            !interaction.replied
+        ) {
+            logger.debug('Responding with error embed');
+            await interaction.deferReply();
+            return await interaction.editReply(errorReply);
+        } else {
+            logger.debug(
+                `${
+                    InteractionType[interaction.type]
+                } interaction '${interactionIdentifier}' threw an error. Cannot send error reply.`
+            );
+
+            if (error.message === 'Unknown interaction') {
+                logger.debug('Interaction no longer exists, timed out, or has already been responded to.');
                 return;
-            case false:
-                logger.debug('Responding with error embed');
-                return await interaction.editReply(errorReply);
-        }
-    } else {
-        logger.debug(
-            `${
-                InteractionType[interaction.type]
-            } interaction '${interactionIdentifier}' threw an error. Cannot send error reply.`
-        );
+            }
 
-        if (error.message === 'Unknown interaction') {
-            logger.debug('Interaction no longer exists, timed out or has already been responded to.');
-            return;
+            if (
+                error.message === 'Collector received no interactions before ending with reason: time' &&
+                interaction instanceof MessageComponentInteraction &&
+                !interaction.replied
+            ) {
+                logger.debug('Interaction collector response timed out.');
+                await interaction.deferReply();
+                return await interaction.editReply(errorReply);
+            }
         }
 
-        if (error.message === 'Collector received no interactions before ending with reason: time') {
-            logger.debug('Interaction collector response timed out.');
-            return;
-        }
+        logger.fatal(
+            error,
+            `Unhandled error while handling interaction '${interactionIdentifier}'. Execution ID: ${executionId}`
+        );
+    } catch (err) {
```

Comment:
Error is already thrown and catched, and only passed on as object into this error handler. The error can no longer crash the program, so a try/catch is unnecessary.

If an error occurs during handling here, e.g. when defering or replying to interaction, then a new error is thrown and will be catched in bot.ts which has it's try/catch block.